src/sidebar/oauth-auth.js

```diff
@@ -16,23 +16,62 @@ function auth($http, settings) {
   var accessTokenPromise;
   var tokenUrl = resolve('token', settings.apiUrl);
 
+  // Return a new tokenInfo object from the given tokenUrl endpoint response.
+  function tokenInfoFrom(response) {
+    var data = response.data;
+    return {
+      accessToken:  data.access_token,
+      expiresIn:    data.expires_in,
```

Comment:
In any variables that stores durations, it is helpful to make sure the units are clear (seconds, mills, nano?). This could be either via naming (eg. `expiresInSecs`) or via documentation of the type somewhere (eg. a JSDoc `@typedef`)